.github/workflows/lint-test-sdk.yml

```diff
@@ -1,13 +1,33 @@
 name: Lint and Test SDK
+
+env:
+  SDK_BRANCH_NAME: ${{ inputs.sdk_branch  || github.head_ref || github.ref_name || 'main' }}
+  TEST_DATA_BRANCH_NAME: ${{ inputs.test_data_branch || 'main' }}
+
 on:
   pull_request:
-    path: '**/*'
-
+    branches: [ "*" ]
+  workflow_dispatch:
+  workflow_call:
+    inputs:
+      test_data_branch:
+        type: string
+        description: The branch in sdk-test-data to target for testcase files
+        required: false
+        default: main
+      sdk_branch:
+        type: string
+        description: The branch of the SDK to test
+        required: false
+          
 jobs:
   lint-test-sdk:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          repository: Eppo-exp/node-server-sdk
```

Comment:
explicitly specify the SDK repository so when this workflow is reused, it's not trying to checkout the sdk-test-data repo here.